### ISSUE TITLE
Getting Values

### ISSUE THREAD

**Issue by alberk8** [AronRubin/SeeedHeartBreathRadar#1]
Hi,

Are you able to get values for CD_HUMAN_PRESENCE, CD_SLEEP_INFO?

**Comment by AronRubin** [comment]
Human presence values are transmitted by the unit and are interpreted as expected. I don't know what would trigger sleep info.

**Comment by alberk8** [comment]
> Human presence values are transmitted by the unit and are interpreted as expected. I don't know what would trigger sleep info.

Is the value that you are getting for the Pulse and Respiratory Rate Consistent?   I am getting values even when there is no human around.

**Comment by AronRubin** [comment]
There are definitely some soft spots in the handling of no person case. I think if there is anything in the frequency range of human heartbeat, even noise, it will show presence.

**Comment by alberk8** [comment]
Ok thank you.  Are you able to do query against the module, like get the firmware value or set values to the unit?

**Comment by AronRubin** [comment]
No, none of that seems to work. I don't know if there is an error in how I composed the frame or if it is unimplemented.

**Comment by DuncanIdahoCT** [comment]
I thought this was just me :) I have the Fall Detection variant of this module and I see only presence and movement bytes, never anything else and I’ve enabled as much raw debug output as I can and logged over time, just to see if I’m missing something or there is a glitch in the Seeed example sketch. I’ve also followed the guide for sending bytes to the module I have and it doesn’t seem to care, I’ve tried both using code inside an Arduino IDE sketch (write.serial) and also just with a comtool debug program and USB UART device so I could see the raw serial responses as documented in the Seeed universal protocol pdf… nothing.

**Comment by AronRubin** [comment]
I am afraid I do not have the Fall variant in hand to test my code with. Note my implementation is not official. I just wanted certain features from a library for this device. I have done my best to pull together all the various documentation sources as a coherent library. It definitely works for the heart and breathing sensor. From discussions with Megdu on the Seeed discord it seems there are a number of mounting parameters to get right for fall detection to work.

**Comment by DuncanIdahoCT** [comment]
Yes I noticed that discussion too or at least one like it. Height, angle, top-down, etc... all play into how well or if at all you'll get fall detection. I loaded up your code onto my MCU with the fall detection module and I see some movement data at least, of course no HR/RR data but it does work with this module at least in part. I did noticed this error during compile however:

error: body of constexpr function 'static constexpr uint8_t SeeedHeartBreathRadar::calcCrc(uint8_t, const uint8_t*, size_t)' not a return-statement

I simply commented out the entire section lol, and it compiles but don't we need that ^ haha

**Comment by DuncanIdahoCT** [comment]
Another question a bit more on-topic... I noticed in your code there is byte evaluation for movement "getmovementstate" but not for presence and I see in the Seeed example code at least, that the Heart Breathe radar should have that base ability just as the Fall Detection one does. I've added the code:

Presence 0)   Movement 0)   HR (unknown) 0  RR (unknown) 0  at 0  angled 0 <-no one / no move
Presence 1)   Movement 2)   HR (unknown) 0  RR (unknown) 0  at 0  angled 0 <-someone / moving
Presence 1)   Movement 1)   HR (unknown) 0  RR (unknown) 0  at 0  angled 0 <-someone / still

It works but, just curious, why it wasn't in yours? I'm really enjoying your code structure and the cleanliness of it by the way!
